src/features/wallet/Wallet.tsx

```diff
@@ -52,18 +52,29 @@ const WrappedWallet: React.FC<Props> = ({
 
   const { walletService } = useContext(WalletContext);
 
-  const { address, chainId, connector } = useAccount();
+  const { address, chainId } = useAccount();
 
   useEffect(() => {
-    if (connector?.name !== "Mobile Browser Provider") {
+    const walletServiceAddress = walletService.state.context.address;
+
+    // Metamask Mobile accidentally triggers this on route changes
+    const didChange = address !== walletServiceAddress;
+    if (
+      didChange &&
+      (
+        ["initialising", "signing", "linking", "minting", "migrating"] as const
+      ).every((state) => !walletState.matches(state))
+    ) {
```

Comment:
This is to stop wallet machine from resetting while we are in the middle of a service. An example of this is when a player connects their wallet in `initialising`. No need to reset the wallet machine as it was us who invoked this. We should only react to wallet changes when the machine is idle.